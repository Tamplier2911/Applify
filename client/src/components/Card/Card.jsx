// import "./Card.scss";
import React from "react";
import PropTypes from "prop-types";

// JS Rendering CSS
import {
  CardContainer,
  CardIconWrapper,
  CardContent,
  CardContentTitle,
  CardContentText,
} from "./CardStyles";

// component motions
import cardMotions from "./CardMotions";

const Card = ({ title, text, gradient, SVG, id }) => {
  const [left, right] = gradient ? gradient : [];
  return (
    <CardContainer
      val={id}
      cl={{ left: left, right: right }}
      animateIn={cardMotions[id]}
      animateOnce={true}
      data-test="card-component"
    >
      <CardIconWrapper data-test="card-component-svg">
        {SVG ? <SVG style={{ fill: left }} /> : null}
      </CardIconWrapper>
      <CardContent data-test="card-component-content">
        <CardContentTitle>{title}</CardContentTitle>
        <CardContentText>{text}</CardContentText>
      </CardContent>
    </CardContainer>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
  gradient: PropTypes.array,
  SVG: PropTypes.object,
  id: PropTypes.number,
};

export default Card;
