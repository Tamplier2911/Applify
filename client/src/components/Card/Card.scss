/*
.home-cards__card {
  display: grid;
  grid-template-rows: 4fr 4fr 1fr;
  position: relative;
  background-position: center;
  background-size: cover;
  min-height: 40rem;

  box-shadow: 0rem 0.1rem 1rem var(--cl-card-shadow);
  transition: box-shadow 0.3s;

  animation: cardFadeIn1 2s;

  &:hover {
    box-shadow: 0rem 0.1rem 1rem var(--cl-card-shadow-hover);
  }

  &:hover .home-cards__card--icon {
    transform: scale(1.4) translateX(-35%);
  }

  &--icon {
    position: absolute;
    width: 8rem;
    height: 8rem;
    background-color: #fff;
    border-radius: 50%;

    top: 13.5rem;
    left: 50%;
    transform: translateX(-50%);

    transition: transform 0.3s;
  }

  &--image {
    position: absolute;
    width: 6rem;
    height: 6rem;
    display: block;
    object-fit: cover;

    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  &--content {
    grid-row: 2 / 3;
    display: grid;
    grid-auto-rows: min-content;
    justify-items: center;
    align-content: center;
    background-color: #fff;
  }

  &--title {
  }

  &--text {
  }
}

@keyframes cardFadeIn1 {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes cardFadeIn2 {
  0% {
    opacity: 0;
  }
  20% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes cardFadeIn3 {
  0% {
    opacity: 0;
  }
  40% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes cardFadeIn4 {
  0% {
    opacity: 0;
  }
  60% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.card-201 {
}
.card-202 {
}
.card-203 {
}
.card-204 {
  @media only screen and (max-width: 1149px) {
    grid-column: 1 / -1;
  }

  @media only screen and (max-width: 864px) {
    grid-column: auto;
  }
}
*/
