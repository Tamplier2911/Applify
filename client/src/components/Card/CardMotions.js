const cardMotions = {
  201: "bounceInLeft",
  202: "flipInY",
  203: "flipInY",
  204: "bounceInRight",
};

// const cardMotionsForFramer = {
//   201: {
//     visible: {
//       opacity: 1,
//       x: 0,
//       y: 0,
//       transition: { duration: 1.2, delay: 0.1, type: "spring", stiffness: 20 },
//     },
//     hidden: { opacity: 0, x: -500 },
//   },
//   202: {
//     visible: {
//       opacity: 1,
//       x: 0,
//       y: 0,
//       transition: { duration: 1.2, delay: 0.1, type: "spring", stiffness: 20 },
//     },
//     hidden: { opacity: 0, y: -300 },
//   },
//   203: {
//     visible: {
//       opacity: 1,
//       x: 0,
//       y: 0,
//       transition: { duration: 1.2, delay: 0.1, type: "spring", stiffness: 20 },
//     },
//     hidden: { opacity: 0, y: 500 },
//   },
//   204: {
//     visible: {
//       opacity: 1,
//       x: 0,
//       y: 0,
//       transition: { duration: 1.2, delay: 0.1, type: "spring", stiffness: 20 },
//     },
//     hidden: { opacity: 0, x: 500 },
//   },
// };

/* <CardContainer
  val={id}
  cl={{ left: gradient[0], right: gradient[1] }}
  initial="hidden"
  animate="visible"
  variants={cardMotions[id]}
></CardContainer>;  */

export default cardMotions;
