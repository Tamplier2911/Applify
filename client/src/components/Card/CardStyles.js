import styled from "styled-components";

const cardSpawnAnimation = props => {
  const { val } = props;
  if (val === 201) {
    return `
        animation: cardFadeIn1 2s;
        @keyframes cardFadeIn1 {
            0% {
              opacity: 0;
            }
            10% {
                opacity: 0;
            }
            100% {
              opacity: 1;
            }
        }
    `;
  } else if (val === 202) {
    return ` 
        animation: cardFadeIn2 2s;
        @keyframes cardFadeIn2 {
            0% {
                opacity: 0;
            }
            20% {
                opacity: 0;
            }
            100% {
                opacity: 1;
            }
        }
      
    `;
  } else if (val === 203) {
    return ` 
        animation: cardFadeIn2 3s;
        @keyframes cardFadeIn2 {
            0% {
                opacity: 0;
            }
            30% {
                opacity: 0;
            }
            100% {
                opacity: 1;
            }
        }
    
    `;
  } else if (val === 204) {
    return ` 
        animation: cardFadeIn2 4s;
        @keyframes cardFadeIn2 {
            0% {
                opacity: 0;
            }
            40% {
                opacity: 0;
            }
            100% {
                opacity: 1;
            }
        }
    
    `;
  }
};

const getCardBackground = props => {
  const {
    cl: { left, right }
  } = props;
  return `background: linear-gradient(to right, ${left}, ${right});`;
};

const getCardRsponsiveness = props => {
  const { val } = props;
  if (val === 204) {
    return `
        @media only screen and (max-width: 1157px) {
            grid-column: 1 / -1;
        }
        @media only screen and (max-width: 872px) {
            grid-column: auto;
        }
    `;
  }
};

export const CardIconWrapper = styled.div`
  position: absolute;
  width: 8rem;
  height: 8rem;
  background-color: var(--cl-white);
  border-radius: 50%;

  top: 13.5rem;
  left: 50%;
  transform: translateX(-50%);

  transition: transform 0.3s;
`;

export const CardContainer = styled.div`
  ${getCardBackground}
  ${getCardRsponsiveness}
  ${cardSpawnAnimation}
  display: grid;
  grid-template-rows: 4fr 4fr 1fr;
  position: relative;
  background-position: center;
  background-size: cover;
  min-height: 40rem;

  box-shadow: 0rem 0.1rem 1rem var(--cl-card-shadow);
  transition: box-shadow 0.3s;

//   animation: cardFadeIn1 2s;

  &:hover {
    box-shadow: 0rem 0.1rem 1rem var(--cl-card-shadow-hover);
  }

  &:hover ${CardIconWrapper} {
    transform: scale(1.4) translateX(-35%);
  }
`;

export const CardContent = styled.div`
  grid-row: 2 / 3;
  display: grid;
  grid-auto-rows: min-content;
  justify-items: center;
  align-content: center;
  background-color: var(--cl-white);
`;

export const CardContentTitle = styled.h4``;

export const CardContentText = styled.div``;
