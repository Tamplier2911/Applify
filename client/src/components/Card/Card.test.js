import React from "react";
import toJson from "enzyme-to-json";

// component
import Card from "./Card";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<Card />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<Card />);
  });

  it("renders without an error", () => {
    const card = findByTestAttr(wrapper, "card-component");
    expect(card.length).toBe(1);
  });

  it("renders child - svg", () => {
    const cardSvg = findByTestAttr(wrapper, "card-component-svg");
    expect(cardSvg.length).toBe(1);
  });

  it("renders child - content", () => {
    const cardContent = findByTestAttr(wrapper, "card-component-content");
    expect(cardContent.length).toBe(1);
  });

  it("recieves expected prop types", () => {
    const expectedProps = {
      title: "str",
      text: "str",
      gradient: [],
      SVG: {},
      id: 777,
    };
    const err = checkProps(Card, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const card = findByTestAttr(wrapper, "card-component");
    expect(toJson(card)).toMatchSnapshot();
  });
});
