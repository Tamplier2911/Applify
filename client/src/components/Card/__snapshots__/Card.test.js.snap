// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
<Styled(ScrollAnimation)
  animateOnce={true}
  cl={
    Object {
      "left": undefined,
      "right": undefined,
    }
  }
  data-test="card-component"
>
  <styled.div
    data-test="card-component-svg"
  />
  <styled.div
    data-test="card-component-content"
  >
    <styled.h4 />
    <styled.div />
  </styled.div>
</Styled(ScrollAnimation)>
`;
